astropy/utils/iers/tests/test_iers.py

```diff
@@ -149,9 +149,16 @@ def test_simple(self):
         assert status3 == iers.FROM_IERS_A_PREDICTION
         assert ut1_utc3 != 0.
 
+
 class TestIERS_Auto():
 
     @remote_data
```

Comment:
So one important thing is the following code in `iers.py`:
```
# If internet is off for testing then do not download.  This
# makes most tests fall back to using the built-in IERS-B table.
conf.auto_download &= not disable_internet.INTERNET_OFF
```
So basically `auto_download` is always False during testing unless `--remote-data` is specified.

I can remove the `@remote_data` and it will fail locally with the unpatched code if I run the testing with `--remote-data` (and pass with the patched code).

@pllim - the example you gave is basically the workaround.  This works because of an "accidental" match here:
```
            if cls.iers_table.meta.get('data_url') == conf.iers_auto_url:
                return cls.iers_table
```
In this case the table meta does not have `data_url` so the `get()` function returns `None`.